packages/language-support/src/autocompletion/schemaBasedCompletions.ts

```diff
@@ -52,78 +52,116 @@ export const allReltypeCompletions = (dbSchema: DbSchema) =>
   reltypesToCompletions(dbSchema.relationshipTypes);
 
 function intersectChildren(
-  connectedLabels: Map<string, Set<string>>,
+  incomingLabels: Map<string, Set<string>>,
+  outGoingLabels: Map<string, Set<string>>,
   children: LabelOrCondition[],
-): Set<string> {
-  let intersection: Set<string> = undefined;
+): { inLabels: Set<string>; outLabels: Set<string> } {
+  let inLabels: Set<string> = undefined;
+  let outLabels: Set<string> = undefined;
   children.forEach((c) => {
-    intersection = intersection
-      ? (intersection = intersection.intersection(
-          walkLabelTree(connectedLabels, c),
-        ))
-      : walkLabelTree(connectedLabels, c);
+    const { inLabels: incoming, outLabels: outgoing } = walkLabelTree(
+      incomingLabels,
+      outGoingLabels,
+      c,
+    );
+    if (!inLabels) {
+      inLabels = incoming;
+      outLabels = outgoing;
+    } else {
+      inLabels = inLabels.intersection(incoming);
+      outLabels = outLabels.intersection(outgoing);
+    }
   });
-  return intersection ?? new Set();
+  if (!inLabels) {
+    inLabels = new Set();
+    outLabels = new Set();
+  }
```

Comment:
Would it make sense to create these at line 59 & 60, and then check "if empty union" otherwise "intersection"? This avoids having to do the null checks and could perhaps be a little clearer